### PR TITLE
FT- Login Endpoint

### PR DESCRIPTION
**Pull request by Hurntre**
#### Description
A Log in endpoint was created in this branch
#### Type of change

- [ ]
- [ ]

#### How Has This Been Tested?
- [ ] integrated testing

#### PT-ID

[](url)

#### Screenshots
image

#### Questions:

### CODE REVIEW

File: server/controllers/auth.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import UserModel from '../db/models/user';
 import tokenGenerator from '../helpers/userToken';
+import authHelper from '../helpers/auth';
 
 const signUpController = (req, res) => {
   const { body } = req;
-  const { id, email } = req;
-  const token = tokenGenerator.userToken({ id, email });
+  const { email } = body;
+  const token = tokenGenerator.userToken({ email });
```
Comment (vic3king) [inline server/controllers/auth.js]: @Hurntre Any reason for this change? It's good practice for user tokens to contain the user's unique ID so we can use it later on in our application.

File: tests/integration/loginValidation.test.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+import seeds from '../../server/db/seeders/seed';
+
+chai.use(chaiHttp);
+
+// should clear and seed db where needed
+before(() => seeds());
+
+describe('User Creation then Login', () => {
+  const signUpRoute = '/api/v1/auth/signup';
+  const loginRoute = '/api/v1/auth/login';
```
Comment (vic3king) [inline tests/integration/loginValidation.test.js]: @Hurntre Nice work coming up with this solution. However, a cleaner approach would be to make use of our seeders file to create a new user in the DB and then use that user to login during testing. Basically, all you need to do in the seed file would be to import the `UserModel`, and then call the create function just like you did in the signup controller and then pass in any dummy user data. the rest has already been wired up... remember this? `before(() => seeds());`

File: server/controllers/auth.js (hunk 1 above)
Comment (Hurntre) [reply server/controllers/auth.js]: there is no ID in the request body. or is there?

File: tests/integration/loginValidation.test.js (hunk 2 above)
Comment (Hurntre) [reply tests/integration/loginValidation.test.js]: i will do that

File: server/controllers/auth.js [hunk 3]
```diff
@@ -13,16 +12,50 @@ const signUpController = (req, res) => {
...
+  const verifyPassword = authHelper.comparePassword(user.password, password);
+
+  if (!verifyPassword) {
+    return res.status(400).send({
+      success: false,
+      error: 'Invalid password',
+    });
+  }
+
+  const token = tokenGenerator.userToken({ email });
```
Comment (vic3king) [inline server/controllers/auth.js]: Please update this line to this `const token = tokenGenerator.userToken({ _id, email });` Missed it the last time

**Pull request merged**